build_tools/cmake/iree_cc_library.cmake

```diff
@@ -115,6 +115,16 @@ function(iree_cc_library)
     list(APPEND _RULE_DEPS ${IREE_IMPLICIT_DEFS_CC_DEPS})
   endif()
 
+  # Find and add threads as dependency.
+  if(NOT ANDROID AND IREE_ENABLE_THREADING)
+    set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+    set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+    find_package(Threads)
```

Comment:
Should we put the `find_package` at a higher level? I don't think we want to repeatedly find the package in each rule creation. I think what we want is for the linkage to `Threads::Threads` to be set here (although actually it would ideally be only on the libraries that actually use threads) but for package discovery to happen once. It's also not clear to me whether projects that depend on us should dictate what threads library gets linked or whether we really want to unconditionally use pthread even when we're a dependency. Can you do what Scott suggested instead? That is, create a variable or target in `iree_copts.cmake` and then use it here. Given that we're already unconditionally using pthread regardless of what a calling project might want, I think it's ok to keep that and we can investigate later if that's really what we want.

Reply:
I have place those piece of code back to `iree_copts.cmake`, so it only find the library once. It will set the `IREE_THREADS_DEPS` once threading is enabled. In `iree_cc_library`, it will link to `IREE_THREADS_DEPS`. Just like what Scott suggested.